scs2-bullet-simulation/src/main/java/us/ihmc/scs2/simulation/bullet/physicsEngine/BulletTools.java

```diff
@@ -31,20 +32,25 @@
 
 public class BulletTools
 {
+   private static final btVector3 origin = new btVector3();
+   private static final btMatrix3x3 basis = new btMatrix3x3();
```

Comment:
I wouldn't recommend doing that. This makes things here thread unsafe and actually these small object are very likely taken from the stack when created inside the method. I know YourKit changes that behavior such that when you profile it, they will show up as being taken from the heap, but this not necessarily true.

Reply:
Yeah that makes sense. I am left still not knowing what cause the heap allocations in the previous implementation. It must be happening on the native side somehow. But in any case, this didn't make a measurable performance impact so it was kind of silly of me to make this PR.